File: programmers/pg_lv2_의상.js
function solution(clothes) {
  let clothesMap = new Map();
  clothes.forEach((ele) => {
    let [name, kind] = ele;

    if (clothesMap.has(kind)) clothesMap.set(kind, clothesMap.get(kind) + 1);
    else clothesMap.set(kind, 1);
  });

  let clothesArr = [...clothesMap];

  let result = clothesArr.reduce((acc, cur) => acc * (cur[1] + 1), 1);

  return result - 1;
}

File: python/programmers/pg_lv2_의상.py
from functools import reduce

def solution(clothes):
    closet = dict()
    for item, kind in clothes:
        if kind in closet: closet[kind] += 1
        else: closet[kind] = 1
        
    itemNums = list(closet.values())
    result = reduce(lambda a, b: a * (b + 1), itemNums, 1) - 1
    return result